### DISCUSSION TITLE
Ward identity in One-Loop QCD Process

### DISCUSSION THREAD

**Discussion by mahdihoumanii** [FeynCalc/feyncalc#284] [Q&A]
Hello! 
I am using the Latest version of FC and FH (10) in development  version. 
I am calculating QCD One-Loop qQ-> tTg amplitude where q/Qis a quark/anti-quark and t/T is a top/anti-top. Since I have a single gluon in my amplitude I should be able to say that the polarization is transverse, and if I replace the polarization vector by its momentum, the amplitude should vanish. Now, I generated the diagrams with both FeynArts and the FH QGraf interface and multiplied by a given tensor. 
(*Some Random SME present in the tree-level amplitude*) 
Tens = Spinor[-Momentum[p4, D], mt, 1] . 
    DiracGamma[Momentum[Polarization[p5, I], D], D] . 
    Spinor[Momentum[p3, D], mt, 1] Spinor[-Momentum[p2, D], 0, 1] . 
    DiracGamma[Momentum[p4, D], D] . Spinor[Momentum[p1, D], 0, 1];

diagsLoop[1] = InsertFields[CreateTopologies[1, 0 -> 5,
    		ExcludeTopologies -> {Tadpoles, WFCorrections}], {} ->
    		{ F[3, {1}], -F[3, {1}], F[3, {3}], -F[3, {3}], V[5]}, 
   InsertionLevel -> {Particles}, Model -> "SMQCD", 
   ExcludeParticles -> {S[_], V[1 | 2 | 3 | 4], F[3, {2}], 
     F[4, {1 | 2 | 3}]}];
diagstree[1] = InsertFields[CreateTopologies[0, 0 -> 5,
    		ExcludeTopologies -> {Tadpoles, WFCorrections}], {} ->
    		{F[3, {1}], -F[3, {1}], F[3, {3}], -F[3, {3}], V[5]}, 
   InsertionLevel -> {Particles}, Model -> "SMQCD",
   		ExcludeParticles -> {S[_], V[1 | 2 | 3 | 4], F[3, {2}], 
     F[4, {1 | 2 | 3}]}];

amploop = FCFAConvert[CreateFeynAmp[diagsLoop[1],
    	Truncated -> False, GaugeRules -> {_FAGaugeXi -> 1}, 
    PreFactor -> 1],
   	IncomingMomenta -> {}, OutgoingMomenta -> {p1, p2, p3, p4, p5},
   	LorentzIndexNames -> {mu10}, DropSumOver -> True,
   	SUNIndexNames -> {a10}, SUNFIndexNames -> {a2, a4, a6, a8}, 
   LoopMomenta -> {l},
   	ChangeDimension -> D, List -> True, SMP -> True, 
   UndoChiralSplittings -> True,
   	FinalSubstitutions -> {SMP["m_u"] -> 0 , SMP["m_t"] -> mt, 
     SMP["g_s"] -> gs}];
FCClearScalarProducts[]; 
SetMandelstam[mand, {p1, p2, p3, p4, p5}, {0, 0, mt, mt, 0}]; 
mandelstam = {mand[1, 5] -> s15, mand[1, 2] -> s12, 
   mand[2, 3] -> s23, mand[3, 4] -> s34, mand[4, 5] -> s45};

(*Here I define projector operators for independent  color structures*) 
ColorFactor[a1_, a2_, b1_, b2_, b3_] := 
 SUNFDelta[a1, a2] SUNTF[{SUNIndex[b1]}, SUNFIndex[b2], SUNFIndex[b3]] 
ColorOps = {ColorFactor[a8, a2, a10, a6, a4], 
   ColorFactor[a8, a6, a10, a2, a4], ColorFactor[a2, a4, a10, a6, a8],
    ColorFactor[a4, a6, a10, a2, a8]};
ColorMat = 
  Table[SUNSimplify[ColorOps[[i]]*ComplexConjugate[ColorOps[[j]]], 
     SUNNToCACF -> False] /. SUNN -> 3, {i, 4}, {j, 4}];
Proj = Inverse[ColorMat] . ComplexConjugate[ColorOps];

ampLoop[1] = 
  SUNSimplify[
    amploop*Proj[[3]] /. Polarization[p5, -I] :> p5 /. Glu5 -> a10 /. 
     Glu5 -> a10, SUNNToCACF -> False] /. SUNN -> 3;
ampLoop[2] = (ampLoop[1]*Tens /. Polarization[p5, I] :> p5 // 
    FermionSpinSum);
ampLoop[3] = ampLoop[2] // DiracSimplify // Simplify;
ampLoop[4] = 
  Monitor[Table[
    TID[ampLoop[3][[i]] /. mandelstam, l, ToPaVe -> True], {i, 1, 
     102}], I];

When I collect all PaVe Functions, all Coefficients of functions vanish and then I am left with a coefficient of the following integrals ((D - 2) A0[mt^2] - B0[mt^2, 0, mt^2]*4 *mt^2) which will not vanish after applying //PaXEvaluate. 
ToBe clear: I tried the same thing on tree-level amplitudes, and it vanished, is there something that could be wrong with how I defined the diagrams, because even with QGRAF FeynHelpers, it gave the same results.

**Comment by vsht** [comment]
Sorry, since some years I don't really have time to debug other people's diagram calculations. If a bug is suspected, I usually ask for the
minimal code excerpt that generates the strange behavior as well as the correct result. But with 100 diagrams it's too much stuff to look into.

I can only recommend to flag the diagrams and try to see if there are gauge invariant subsets among them as well as the origin
for the noncancellation. This should be a good starting point to investigate further. Good luck!
